docs/best-practices/index.md

```diff
@@ -0,0 +1,438 @@
+
+Committing secrets, such as passwords, API keys, and tokens, directly into Terraform configurations and subsequently into source control, poses significant security risks that can have far-reaching consequences. This practice exposes sensitive information to anyone who has access to the repository, including potential unauthorized users, thereby compromising the security of your infrastructure. Once secrets are committed to a repository, removing them requires extensive effort and does not guarantee that they haven't been copied or logged elsewhere.
+
+Moreover, version control systems are designed to track and preserve history, making it challenging to completely erase secrets once they are committed. This persistence in history means that even if the secrets are later removed from the codebase, they remain accessible in the commit history. Additionally, repositories, especially public or shared ones, are often cloned, forked, or integrated with third-party services, further increasing the exposure of secrets.
+
+Ultimately the safest way to recover from secrets that have been leaked is to rotate them in the source system, which can be an impactful activity if other systems or individuals depend on that credential.
+
+To mitigate these risks, it's recommended to use secure secrets management tools and practices. Terraform supports various mechanisms for securely managing secrets, including environment variables, encrypted state files, and integration with dedicated secrets management systems like AWS Secrets Manager, Azure Key Vault, or HashiCorp Vault. These tools provide controlled access to secrets, audit trails, and the ability to rotate secrets periodically or in response to a breach.
+
+By keeping secrets out of source control and employing robust secrets management strategies, users can significantly enhance the security posture of their infrastructure deployments. This approach not only protects sensitive information but also aligns with compliance requirements and best practices for secure infrastructure management.
+
+## Multi-team Development
+
+### Use "On-Demand" Development Environments
+
+The recommended approach for multi-team development, when using a GitOps CICD promotion process, is to spin up "on-demand" development and test environments (where possible), specific to new features or to individual teams, to allow for development and integration testing that doesn't conflict with other team's development and test activities.  The Terraform provider allows administrators to use CICD automation to provision new environments as required, and remove them once the project activity no longer needs them.
```

Comment:
When using a GitOps CICD promotion process across multiple teams or individuals, a recommended approach is to spin up "on-demand" development and test environments (where possible).  These environments can be specific to new features or to individual teams to allow for development and integration testing that does not conflict with other team's activities.  The Terraform provider allows administrators to use CICD automation to provision new environments as required, and remove them after the project activity no longer needs them.